contributors/design-proposals/sig-storage/volume-topology-scheduling.md

```diff
@@ -0,0 +1,580 @@
+performance, we’ll assume that the binding will likely succeed, and update the
+PV cache first.  Then the actual binding API update will be made
+asynchronously, and the scheduler can continue processing other Pods.
+
+For the alpha phase, the AssumePVCs function will be directly called by the
+scheduler.  We’ll consider creating a generic scheduler interface in a
+subsequent phase.
+
+```
+AssumePVCs(pod *v1.Pod, node *v1.Node) (pvcBindingRequired bool, err error)
+```
+1. If all the Pod’s PVCs are bound, return false.
+2. For static PV binding:
+    1. Get the cached matching PVs for the PVCs on that Node.
+    2. Validate the actual PV state.
+    3. Mark PV.ClaimRef in the PV cache.
+3. For in-tree and external dynamic provisioning:
+    1. Nothing.
+4. Return true.
+
+
+##### Bind
+If AssumePVCs returns pvcBindingRequired, then the BindPVCs function is called
+as a go routine.  Otherwise, we can continue with assuming and binding the Pod
+to the Node.
+
+For the alpha phase, the BindUnboundPVCs function will be directly called by the
+scheduler.  We’ll consider creating a generic scheduler interface in a subsequent
+phase.
+
+```
+BindUnboundPVCs(pod *v1.Pod, node *v1.Node) (err error)
+```
+1. For static PV binding:
+    1. Prebind the PV by updating the `PersistentVolume.ClaimRef` field.
+    2. If the prebind fails, revert the cache updates.
+    3. Otherwise, wait for the PVCs to be bound, PVC/PV object is deleted, or
+PV.ClaimRef field is cleared
+2. For in-tree dynamic provisioning:
+    1. Make the provision call, which will create a new PV object that is
+prebound to the PVC.
+    2. If provisioning is successful, wait for the PVCs to be bound, PVC/PV
+object is deleted, or PV.ClaimRef field is cleared
+3. For external provisioning:
+    1. Set the annotation for external provisioners.
+4. Send Pod back through scheduling, regardless of success or failure.
+    1. In the case of success, we need one more pass through the scheduler in
+order to evaluate other volume predicates that require the PVC to be bound, as
+described below.
+    2. In the case of failure, we want to retry binding/provisioning.
+
+Note that for external provisioning, we do not wait for the PVCs to be bound, so
+the Pod will be sent through scheduling repeatedly until the PVCs are bound.
+This is because there is no function call for external provisioning, so if we
+did wait, we could be waiting forever for the PVC to bind.  It’s possible that
+in the meantime, a user could create a PV that satisfies the PVC and doesn’t
+need the external provisioner anymore.
+
+TODO: pv controller has a high resync frequency, do we need something similar
+for the scheduler too
+
+##### Pod preemption considerations
+The MatchUnboundPVs predicate does not need to be re-evaluated for pod
+preemption.  Preempting a pod that uses a PV will not free up capacity on that
+node because the PV lifecycle is independent of the Pod’s lifecycle.
+
+##### Other scheduler predicates
```

Comment:
Could you please also evaluation if this new predicate need special care in `factory.go` according to: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go#L70-L76

Reply:
Cool, thanks!